app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -283,6 +283,10 @@ protected void initializeWidgetPlacement(WidgetPlacement aPlacement) {
     }
 
     void setupListeners(Session aSession) {
+        if (aSession != mSession) {
+            Log.w(LOGTAG, "Tried to set up listeners on a session that is not the current one.");
+            return;
+        }
```

Comment:
Is that even possible? If not it should be better to assert as it will unveil a more important issue

Reply:
It is not possible in the current code. I have replaced this with an assert, just in case.